src/tests/suite/daos_obj.c

```diff
@@ -4667,14 +4667,23 @@ int
 run_daos_io_test(int rank, int size, int *sub_tests, int sub_tests_size)
 {
 	int rc = 0;
+	char oclass[16] = {0};
+	char buf[32];
 
 	MPI_Barrier(MPI_COMM_WORLD);
 	if (sub_tests_size == 0) {
 		sub_tests_size = ARRAY_SIZE(io_tests);
 		sub_tests = NULL;
 	}
 
-	rc = run_daos_sub_tests("DAOS_IO", io_tests,
+	if (dt_obj_class != OC_UNKNOWN) {
+		oclass[0] = '_';
+		daos_oclass_id2name(dt_obj_class, &oclass[1]);
+	}
+	snprintf(buf, sizeof(buf), "DAOS_IO%s", oclass);
+	buf[sizeof(buf) - 1] = 0;
```

Comment:
Could there be garbage in buf between "DAOS_IO" and sizeof(buf) -1? Should buf be initialized to {0} like oclass is?

Reply:
No, this is only needed if the string gets truncated.  Otherwise, it doesn't matter that there is garbage because the snprintf will have null terminated it